packages/content-slider/src/components/content-slider/content-slider.component.ts

```diff
@@ -0,0 +1,280 @@
+import { TemplatesComponent } from "@ribajs/core";
+import { throttle } from "@ribajs/utils/src/control";
+import { hasChildNodesTrim } from "@ribajs/utils/src/dom";
+
+import template from "./content-slider.component.html";
+
+export interface SlideItem {
+  /** Template content is the item capture */
+  content: string;
+  imageSrc: string;
+  active: boolean;
+  index: number;
+}
+
+interface Scope {
+  items: SlideItem[];
+  currentIndex: number;
+  activeItem: SlideItem | null;
+  transform: string;
+  previous: ContentSliderComponent["previous"];
+  next: ContentSliderComponent["next"];
+  goTo: ContentSliderComponent["goTo"];
+  activeItemWidth: number;
+  inactiveItemWidth: number;
+  activeClass: string;
+  activeColumnClasses: string[];
+  inactiveColumnClasses: string[];
+}
+
+export class ContentSliderComponent extends TemplatesComponent {
+  public static tagName = "content-slider";
+
+  protected autobind = true;
+
+  protected templateAttributes = [
+    {
+      name: "image-src",
+      required: true,
+    },
+    {
+      name: "active",
+      required: false,
+    },
+    {
+      name: "index",
+      required: false,
+    },
+  ];
+
+  static get observedAttributes() {
+    return ["active-class", "active-column-classes", "inactive-column-classes"];
+  }
+
+  protected scope: Scope = {
+    items: [],
+    currentIndex: 0,
+    activeItem: null,
+    transform: "translateX(0)",
+    previous: this.previous,
+    next: this.next,
+    goTo: this.goTo,
+    activeItemWidth: 0,
+    inactiveItemWidth: 0,
+    activeClass: "active",
+    activeColumnClasses: [
+      "col-10",
+      "col-sm-10",
+      "col-md-6",
+      "col-lg-5",
+      "col-xl-4",
+    ],
+    inactiveColumnClasses: [
+      "col-6",
+      "col-sm-6",
+      "col-md-2",
+      "col-lg-2",
+      "col-xl-2",
+    ],
+  };
+
+  constructor(element?: HTMLElement) {
+    super(element);
+    this.debug("constructor", this);
+  }
+
+  protected connectedCallback() {
+    super.connectedCallback();
+    return this.init(ContentSliderComponent.observedAttributes);
+  }
+
+  protected initItems() {
+    const items = this.el.querySelectorAll(".item");
+    if (!items) {
```

Comment:
```suggestion
    if (items.length === 0) {
```

`items` ist eine NodeList, die leer sein kann (length === 0), aber nicht falsy